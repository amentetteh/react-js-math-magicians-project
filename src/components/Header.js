import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <nav>
      <div id="logo">Math Magicians</div>
      <div className="menu">
        <ul>
          <li className="link"><NavLink className="nav-link" to="/">Home</NavLink></li>
          <li className="link"><NavLink className="nav-link" to="/calculator">Calculator</NavLink></li>
          <li className="link"><NavLink className="nav-link" to="/quote">Quote</NavLink></li>
        </ul>
      </div>
    </nav>
  );
}

export default Header;
