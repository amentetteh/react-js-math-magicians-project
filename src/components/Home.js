function Home() {
  return (
    <div className="home">
      <h3>Welcome to our Page</h3>

      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut non quam a risus
        eleifend tincidunt. Maecenas rhoncus porta ex, at sodales tortor consequat
        vitae. Aliquam fringilla lacus elit, vel eleifend lorem sodales eget.
        Sed a euismod mi, eget lacinia dolor. Nulla egestas lorem ut
        ligula auctor, et fermentum nibh commodo. Pellentesque ut purus accumsan,
        aliquam ligula quis, dictum erat. Ut vitae dignissim quam.
        Curabitur nulla enim, interdum vel nibh volutpat, porttitor
        viverra turpis. Donec eu libero sit amet sapien ornare aliquet sed vel erat.
        Donec ultricies, leo sed ullamcorper mollis, est erat
        pharetra augue, eget consequat ipsum lacus in sapien.
        In bibendum et sapien a facilisis. Quisque lacinia mi sit
        amet finibus eleifend. Maecenas sit amet rhoncus mauris.
        Suspendisse tempor lacus non sapien venenatis dapibus.
        Sed placerat ac velit non suscipit. Cras id maximus libero.
      </p>
      <p>
        Pellentesque ullamcorper tempus neque in vestibulum. Pellentesque
        sit amet urna erat. Vestibulum pellentesque lectus id ipsum
        semper, a venenatis erat feugiat. Aliquam porta iaculis ligula
        id mollis. Aenean id pellentesque risus. Morbi lobortis fermentum
        rhoncus. Etiam efficitur vitae eros ac dictum.
        Integer id erat sit amet neque maximus hendrerit. Maecenas quis massa nec leo tincidunt
        convallis. In consequat aliquet neque, nec semper metus vehicula nec. Donec tincidunt
        ligula nec fermentum feugiat. Orci varius natoque penatibus et magnis dis parturient
        montes, nascetur ridiculus mus.
      </p>
    </div>
  );
}

export default Home;
