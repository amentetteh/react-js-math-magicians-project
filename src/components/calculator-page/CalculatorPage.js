import Calculator from './Calculator';

function CalculatorPage() {
  return (
    <div id="calculator-page">
      <aside className="page-left">
        Let us do some maths!
      </aside>
      <aside className="page-right">
        <Calculator />
      </aside>
    </div>
  );
}

export default CalculatorPage;
