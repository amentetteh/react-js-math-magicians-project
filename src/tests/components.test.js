import React from 'react';
import {render} from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import Quote from '../components/Quote';
import Home from '../components/Home';
import CalculatorPage from '../components/calculator-page/CalculatorPage';

describe('Test for operation', () => {
  it('Home component should render correctly', () => {
    const component = render(<Home />);
    expect(component).toMatchSnapshot();
  });
  it('Calculator Page component should render correctly', () => {
    const component = render(<CalculatorPage />);
    expect(component).toMatchSnapshot();
  });
  it('Header component should render correctly', () => {
    const component = render(<Router><Header /></Router>);
    expect(component).toMatchSnapshot();
  });
  it('Quote component should render correctly', () => {
    const component = render(<Quote />);
    expect(component).toMatchSnapshot();
  });
});
