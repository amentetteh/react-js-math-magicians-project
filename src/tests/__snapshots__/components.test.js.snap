// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for operation CalculatorPage component should render correctly 1`] = `
<div
  id="calculator-page"
>
  <aside
    className="page-left"
  >
    Let us do some maths!
  </aside>
  <aside
    className="page-right"
  >
    <div
      id="calculator"
    >
      <div
        className="touch-row"
      >
        <div
          id="screen"
        >
          0
        </div>
      </div>
      <div
        className="touch-row"
      >
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            AC
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            +/-
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            %
          </span>
        </button>
        <button
          className="touch fn-col"
          onClick={[Function]}
          type="button"
        >
          <span>
            รท
          </span>
        </button>
      </div>
      <div
        className="touch-row"
      >
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            7
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            8
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            9
          </span>
        </button>
        <button
          className="touch fn-col"
          onClick={[Function]}
          type="button"
        >
          <span>
            x
          </span>
        </button>
      </div>
      <div
        className="touch-row"
      >
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            4
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            5
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            6
          </span>
        </button>
        <button
          className="touch fn-col"
          onClick={[Function]}
          type="button"
        >
          <span>
            -
          </span>
        </button>
      </div>
      <div
        className="touch-row"
      >
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            1
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            2
          </span>
        </button>
        <button
          className="touch"
          onClick={[Function]}
          type="button"
        >
          <span>
            3
          </span>
        </button>
        <button
          className="touch fn-col"
          onClick={[Function]}
          type="button"
        >
          <span>
            +
          </span>
        </button>
      </div>
      <div
        className="touch-row"
      >
        <div
          className="bottom-line-item"
        >
          <button
            className="touch zero-touch"
            onClick={[Function]}
            type="button"
          >
            <span>
              0
            </span>
          </button>
        </div>
        <div
          className="bottom-line-item right"
        >
          <button
            className="touch"
            onClick={[Function]}
            type="button"
          >
            <span>
              .
            </span>
          </button>
          <button
            className="touch fn-col"
            onClick={[Function]}
            type="button"
          >
            <span>
              =
            </span>
          </button>
        </div>
      </div>
    </div>
  </aside>
</div>
`;

exports[`Test for operation Header component should render correctly 1`] = `
<nav>
  <div
    id="logo"
  >
    Math Magicians
  </div>
  <div
    className="menu"
  >
    <ul>
      <li
        className="link"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          className="nav-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="link"
      >
        <a
          className="nav-link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Test for operation Home component should render correctly 1`] = `
<div
  className="home"
>
  <h3>
    Welcome to our Page
  </h3>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut non quam a risus eleifend tincidunt. Maecenas rhoncus porta ex, at sodales tortor consequat vitae. Aliquam fringilla lacus elit, vel eleifend lorem sodales eget. Sed a euismod mi, eget lacinia dolor. Nulla egestas lorem ut ligula auctor, et fermentum nibh commodo. Pellentesque ut purus accumsan, aliquam ligula quis, dictum erat. Ut vitae dignissim quam. Curabitur nulla enim, interdum vel nibh volutpat, porttitor viverra turpis. Donec eu libero sit amet sapien ornare aliquet sed vel erat. Donec ultricies, leo sed ullamcorper mollis, est erat pharetra augue, eget consequat ipsum lacus in sapien. In bibendum et sapien a facilisis. Quisque lacinia mi sit amet finibus eleifend. Maecenas sit amet rhoncus mauris. Suspendisse tempor lacus non sapien venenatis dapibus. Sed placerat ac velit non suscipit. Cras id maximus libero.
  </p>
  <p>
    Pellentesque ullamcorper tempus neque in vestibulum. Pellentesque sit amet urna erat. Vestibulum pellentesque lectus id ipsum semper, a venenatis erat feugiat. Aliquam porta iaculis ligula id mollis. Aenean id pellentesque risus. Morbi lobortis fermentum rhoncus. Etiam efficitur vitae eros ac dictum. Integer id erat sit amet neque maximus hendrerit. Maecenas quis massa nec leo tincidunt convallis. In consequat aliquet neque, nec semper metus vehicula nec. Donec tincidunt ligula nec fermentum feugiat. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </p>
</div>
`;

exports[`Test for operation Quote component should render correctly 1`] = `
<div
  className="quote"
>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.- William Paul Thurston
  </p>
</div>
`;
